weak_supervision/models/latent_alignment.py

```diff
@@ -16,15 +17,24 @@ class LatentAlignment(Model):
     def __init__(self, vocab: Vocabulary,
                  utterance_embedder: TextFieldEmbedder,
                  logical_form_embedder: TextFieldEmbedder,
+                 utterance_encoder: Seq2SeqEncoder,
                  initializer: InitializerApplicator = InitializerApplicator(),
                  regularizer: Optional[RegularizerApplicator] = None) -> None:
         super().__init__(vocab, regularizer)
 
         self.utterance_embedder = utterance_embedder
         self.logical_form_embedder = logical_form_embedder
-        self.translation_layer = Linear(self.logical_form_embedder.get_output_dim(),
-                                        self.utterance_embedder.get_output_dim())
+        self.utterance_encoder = utterance_encoder
 
+        self.translation_layer = Linear(self.logical_form_embedder.get_output_dim(),
+                                        self.utterance_encoder.get_output_dim())
+
+        self.mean_ranks = 0.0
```

Comment:
You can define all these metrics as instances of `Average` (https://github.com/allenai/allennlp/blob/master/allennlp/training/metrics/average.py), and just call each of the metrics with the value you need to add to the numerator.

Reply:
Thanks! I'll add this once i've done trying out the things we discussed in our meeting